components/homme/src/theta-l_kokkos/cxx/HyperviscosityFunctorImpl.cpp

```diff
@@ -65,13 +73,41 @@ void HyperviscosityFunctorImpl::init_params(const SimulationParams& params)
     ExecViewManaged<Scalar[NUM_LEV]>::HostMirror h_nu_scale_top;
     h_nu_scale_top = Kokkos::create_mirror_view(m_nu_scale_top);
 
-    Kokkos::Array<Real,NUM_BIHARMONIC_PHYSICAL_LEVELS> lev_nu_scale_top = { 4.0, 2.0, 1.0 };
-    for (int phys_lev=0; phys_lev<NUM_BIHARMONIC_PHYSICAL_LEVELS; ++phys_lev) {
+    const auto etai_h = Kokkos::create_mirror_view(m_hvcoord.etai);
+    const auto etam_h = Kokkos::create_mirror_view(m_hvcoord.etam);
+    Kokkos::deep_copy(etai_h, m_hvcoord.etai);
+    Kokkos::deep_copy(etam_h, m_hvcoord.etam);
+
+    for (int phys_lev=0; phys_lev < NUM_LEV*VECTOR_SIZE; ++phys_lev) {
```

Comment:
Since you don't care about the padding at the end, why not looping from 0 to `NUM_PHYSICAL_LEV` at this point?

Reply:
Ah, I see, you want to avoid NaN's in the padding. You could still loop to NUM_PHYSICAL_LEV (avoiding the if in the block), and then _after_ do a for loop on `[NUM_PHYSICAL_LEV,NUM_LEV*VECTOR_SIZE)`, to zero out the padding. Tbc, this is not for perf, it's just to make the body of the for loop a tad less involved (by removing the if/else)